dashboard/src/components/Header/ContextSelector.test.tsx

```diff
@@ -201,18 +198,17 @@ it("changes the location with the new namespace", () => {
         .prop("onClick") as any
     )();
   });
-  expect(push).toHaveBeenCalledWith("/c/default-cluster/ns/other/catalog");
+  expect(history.location.pathname).toBe("/c/default-cluster/ns/other/catalog");
 });
 
 it("changes the location with the new cluster and namespace", () => {
-  const push = jest.fn();
-  spyOnUseHistory = jest.spyOn(ReactRouter, "useHistory").mockReturnValue({ push } as any);
-  spyOnUseLocation = jest.spyOn(ReactRouter, "useLocation").mockReturnValue({
-    pathname: "/c/default-cluster/ns/ns-bar/catalog",
-    search: "",
-    state: "",
-  } as any);
-  const wrapper = mountWrapper(defaultStore, <ContextSelector />);
+  const history = createMemoryHistory({ initialEntries: ["/c/default-cluster/ns/ns-bar/catalog"] });
```

Comment:
An alternative way to the <MemoryRouter initialEntries=......> which allows using the variable to perform the checks